### DISCUSSION TITLE
`redeclaration of formal parameter module` when trying to run tests

### DISCUSSION THREAD

**Discussion by jasdom** [iTwin/iTwinUI#2123] [Q&A] [answered]
Hi, after upgrading to `@itwin/itwinui-react` 3.12.0, our tests will no longer run with the error `SyntaxError: redeclaration of formal parameter module`. We are running our tests using mocha and it seems to fail while trying to compile `itwinui-react/src/utils/meta.ts` due to declaration of `module`.

I have reproduced the issue in a sandbox - https://stackblitz.com/edit/github-sfphuz-r7uo5e?file=package.json. It can be seen after opening a terminal and running `npm run test:cjs`, however the issue goes away if `"type": "module"` is added to `package.json` and `npm run test:esm` is used instead.

Is this a bug or should it be resolved on our side in some way?

**Comment by mayank99** [comment, marked as answer]
Hey, thanks for reporting this. It was a bug on our end that has been fixed in [3.12.1](https://github.com/iTwin/iTwinUI/releases/tag/%40itwin%2Fitwinui-react%403.12.1).

You should probably still use ESM, but now CJS will work fine.
